lib/rules/component-defined-inline.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "Ensure component object is defined inline, and not created as a class or named object",
+      category: "Possible Errors",
+      recommended: true
+    },
+    schema: []
+  },
+  create: (context) => {
+    return {
+      'MemberExpression': (node) => {
+        if (node.object.name !== 'angular' && node.property.name !== 'module') {
+          return
+        }
```

Comment:
That seems like it could be an improvement.  This type of code does seem to be like regex programming, where it gets unreadable pretty quickly...